src/freeze/enable.rs

```diff
@@ -0,0 +1,127 @@
+use super::*;
+
+pub struct EnableFreezeArgs {
+    pub keypair: Option<String>,
+    pub rpc_url: Option<String>,
+    pub cache: String,
+    pub config: String,
+    pub candy_machine: Option<String>,
+    pub freeze_days: Option<u8>,
+}
+
+pub fn process_enable_freeze(args: EnableFreezeArgs) -> Result<()> {
+    let sugar_config = sugar_setup(args.keypair.clone(), args.rpc_url.clone())?;
+    let client = setup_client(&sugar_config)?;
+    let program = client.program(CANDY_MACHINE_ID);
+    let config_data = get_config_data(&args.config)?;
+
+    // The candy machine id specified takes precedence over the one from the cache.
+    let candy_machine_id = match args.candy_machine {
+        Some(ref candy_machine_id) => candy_machine_id.to_owned(),
+        None => {
+            let cache = load_cache(&args.cache, false)?;
+            cache.program.candy_machine
+        }
+    };
+
+    let candy_pubkey = Pubkey::from_str(&candy_machine_id)
+        .map_err(|_| anyhow!("Failed to parse candy machine id: {}", &candy_machine_id))?;
+
+    // Freeze days specified takes precedence over the one from the config.
+    let freeze_time = if let Some(freeze_days) = args.freeze_days {
+        (freeze_days * 24 * 60 * 60) as i64
+    } else if let Some(freeze_time) = config_data.freeze_time {
+        freeze_time
+    } else {
+        return Err(anyhow!(
+            "No freeze time specified either in config or as argument"
+        ));
+    };
+
+    println!(
+        "{} {}Loading candy machine",
+        style("[1/2]").bold().dim(),
+        LOOKING_GLASS_EMOJI
+    );
+    println!("{} {}", style("Candy machine ID:").bold(), candy_machine_id);
+
+    let pb = spinner_with_style();
+    pb.set_message("Connecting...");
+    let candy_machine_state =
+        get_candy_machine_state(&sugar_config, &Pubkey::from_str(&candy_machine_id)?)?;
+
+    pb.finish_with_message("Done");
+
+    assert_correct_authority(
+        &sugar_config.keypair.pubkey(),
+        &candy_machine_state.authority,
+    )?;
+
+    // Cannot enable freeze if minting has started.
+    if candy_machine_state.items_redeemed > 0 {
+        return Err(anyhow!("Cannot enable freeze after minting has started"));
+    }
+
+    println!(
+        "\n{} {}Turning on freeze feature for candy machine",
+        style("[2/2]").bold().dim(),
+        ICE_CUBE_EMOJI
+    );
+
+    let pb = spinner_with_style();
+    pb.set_message("Sending Enable Freeze transaction...");
+
+    let signature = enable_freeze(&program, &config_data, &candy_pubkey, freeze_time)?;
```

Comment:
Similar to the disable, trying to enable an already enabled freeze candy machine gives an unclear error:
<img width="682" alt="image" src="https://user-images.githubusercontent.com/729235/190192776-26b82e4b-866a-4321-888a-5a6d1bb87701.png">

Reply:
[Fixed.](https://github.com/metaplex-foundation/sugar/pull/345/commits/a4d09052c789a0156cc8cd9f3fd3ed4d7d0235ff)

![Selection_962](https://user-images.githubusercontent.com/1684605/190213670-ff7b55a8-7a06-4465-92e1-466e76f6d08c.png)